### PR TITLE
Improve Data.MinLen documentation

### PR DESCRIPTION
**Pull request by kbillings**

### CODE REVIEW

File: src/Data/MinLen.hs [hunk 1]
```diff
@@ -16,7 +16,7 @@ module Data.MinLen
     , TypeNat (..)
     , AddNat
     , MaxNat
-      -- * Minimum length newtype wrapper
+      -- * Minimum length data wrapper
```
Comment (snoyberg) [inline src/Data/MinLen.hs]: What's the motivation for this change? It seems like it's an accurate description to call it a newtype wrapper.

**Comment by snoyberg** [comment]
@MaxGabriel If I'm not mistaken, you recently worked on this documentation, correct?

File: src/Data/MinLen.hs (hunk 1 above)
Comment (kbillings) [inline src/Data/MinLen.hs]: Sorry, messed up I'll revert it.

File: src/Data/MinLen.hs [hunk 2]
```diff
@@ -149,24 +203,32 @@ natProxy _ = typeNat
 --
 -- ==== __Examples__
 --
--- > > 1 `mlcons` toMinLenZero []
--- > MinLen {unMinLen = [1]}
+-- @
+-- > > 1 \`mlcons` 'toMinLenZero' []
+-- > 'MinLen' {unMinLen = [1]}
+-- @
```
Comment (MaxGabriel) [inline src/Data/MinLen.hs]: This can have the extra `>` removed since you're using `@` blocks

**Comment by MaxGabriel** [comment]
Yeah, these changes look good to me.

**Comment by MaxGabriel** [comment]
Not sure what others think about this, but do you feel like the emphasis on **safe** improves the documentation? It seems safe is a reasonable default, especially for this module, and possibly steals attention from the other two bolded things (**Unsafe** and function names). 

Here's a comparison:
* [Bolded](http://maxgabriel.github.io/bolded/Data.MinLen.html)
* [Regular](http://maxgabriel.github.io/regular/Data.MinLen.html)

**Comment by gregwebs** [comment]
I do agree that in Haskell every function should be assumed safe by default.

**Comment by kbillings** [comment]
Ok I'll remove the emphasis on the safe functions.
